torchvision/models/maxvit.py

```diff
@@ -0,0 +1,595 @@
+        # conv parameters
+        in_channels: int,
+        out_channels: int,
+        squeeze_ratio: float,
+        expansion_ratio: float,
+        # conv + transformer parameters
+        normalization_fn: Callable[..., nn.Module],
+        activation_fn: Callable[..., nn.Module],
+        # transformer parameters
+        head_dim: int,
+        mlp_ratio: int,
+        mlp_dropout: float,
+        attn_dropout: float,
+        # partitioning parameters
+        partition_size: int,
+        input_grid_size: Tuple[int, int],
+        # number of layers
+        n_layers: int,
+        p_stochastic: List[float],
+    ) -> None:
+        super().__init__()
+        assert (
+            len(p_stochastic) == n_layers
+        ), f"p_stochastic must have length n_layers={n_layers}, got p_stochastic={p_stochastic}."
+
+        self.layers = nn.ModuleList()
+        # account for the first stride of the first layer
+        self.grid_size = (input_grid_size[0] // 2, input_grid_size[1] // 2)
+
+        for idx, p in enumerate(p_stochastic):
+            stride = 2 if idx == 0 else 1
+            self.layers += [
+                MaxVitLayer(
+                    in_channels=in_channels if idx == 0 else out_channels,
+                    out_channels=out_channels,
+                    squeeze_ratio=squeeze_ratio,
+                    expansion_ratio=expansion_ratio,
+                    stride=stride,
+                    normalization_fn=normalization_fn,
+                    activation_fn=activation_fn,
+                    head_dim=head_dim,
+                    mlp_ratio=mlp_ratio,
+                    mlp_dropout=mlp_dropout,
+                    attn_dropout=attn_dropout,
+                    partition_size=partition_size,
+                    grid_size=self.grid_size,
+                ),
+                StochasticDepth(p, mode="row"),
+            ]
+
+    def forward(self, x: Tensor) -> Tensor:
+        for layer in self.layers:
+            x = layer(x)
+        return x
+
+
+class MaxVit(nn.Module):
+    def __init__(
+        self,
+        # stem and task parameters
+        input_channels: int,
+        stem_channels: int,
+        input_size: Tuple[int, int],
+        out_classes: int,
+        # block parameters
+        block_channels: List[int],
+        block_layers: List[int],
+        stochastic_depth_prob: float,
+        # conv parameters
+        squeeze_ratio: float,
+        expansion_ratio: float,
+        # conv + transformer parameters
+        # normalization_fn is applied only to the conv layers
+        # activation_fn is applied both to conv and transformer layers
+        normalization_fn: Callable[..., nn.Module],
+        activation_fn: Callable[..., nn.Module],
+        # transformer parameters
+        head_dim: int,
+        mlp_ratio: int,
+        mlp_dropout: float,
+        attn_dropout: float,
+        # partitioning parameters
+        partition_size: int,
+    ) -> None:
+        super().__init__()
```

Comment:
You need at least a logger here. Maybe in other blocks that you plan to expose on the API.
```suggestion
        super().__init__()
        _log_api_usage_once(self)
```